cmd/rpcdaemon/commands/debug_api.go

```diff
@@ -118,33 +117,45 @@ func (api *PrivateDebugAPIImpl) AccountRange(ctx context.Context, blockNrOrHash
 	return res, nil
 }
 
-func (api *PrivateDebugAPIImpl) GetModifiedAccountsByNumber(_ context.Context, startNum rpc.BlockNumber, endNum *rpc.BlockNumber) ([]common.Address, error) {
-	if endNum != nil && startNum.Int64() >= endNum.Int64() {
-		return nil, fmt.Errorf("start block height (%d) must be less than end block height (%d)", startNum.Int64(), endNum.Int64())
+// GetModifiedAccountsByNumber returns a list of accounts found in the change sets
+// startNum - first block from which to include results
+// endNum - if present, last block from which to include results (inclusive). If not present, startNum.
+func (api *PrivateDebugAPIImpl) GetModifiedAccountsByNumber(ctx context.Context, startNumber rpc.BlockNumber, endNumber *rpc.BlockNumber) ([]common.Address, error) {
+	tx, err := api.dbReader.Begin(ctx)
```

Comment:
I suggest using `tx, err := api.db.Begin(ctx, nil, false)` instead. Second argument is parent (`nil`), third argument is writable (`false`). Otherwise, `Begin` looks like we want a writeable transaction, and only getting a read-only one because we are remote. But if running against local chaindata, it will be actually writeable. Further on in this function, you should try to use `tx` instead of `api.dbReader`

Reply:
This same pattern appears in a lot of other places in the RPC code. Should I change them everywhere?

I can make another issue and PR to track that change or do it as part of this PR if you wish.

Reply:
If you can, yes, as long as there is a way to test this. This may improve performance, because we do not end up opening transactions multiple times